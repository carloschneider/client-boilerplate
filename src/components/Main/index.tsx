import * as S from './styles'

export type MainProps = {
  title: string
}

const Main = ({ title }: MainProps) => (
  <S.Wrapper>
    <h1>{title}</h1>
  </S.Wrapper>
)

export default Main
