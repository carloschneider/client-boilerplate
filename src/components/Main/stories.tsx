import { Story, Meta } from '@storybook/react'

import Main, { MainProps } from '.'

export default {
  title: 'Main',
  component: Main,
} as Meta

const Template: Story<MainProps> = (args) => <Main {...args} />

export const Default = Template.bind({})
Default.args = {
  title: 'Nice',
}
