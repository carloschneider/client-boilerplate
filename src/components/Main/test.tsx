import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should be render the component', () => {
    const { container } = render(<Main title="Hello world" />)

    expect(
      screen.getByRole('heading', { name: /hello world/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
