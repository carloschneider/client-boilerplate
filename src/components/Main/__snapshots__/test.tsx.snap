// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be render the component 1`] = `
<h1>
  Hello world
</h1>
`
